components/MashDashboard/MashDashboard.spec.tsx

```diff
@@ -0,0 +1,74 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import { mockedMashReferrals } from 'factories/mashReferral';
+import MashDashboard from './MashDashboard';
+
+let tabType: string | undefined;
+let confirmationType: string | undefined;
+const mockedRouter = {
+  query: { confirmation: confirmationType, tab: tabType },
+  push: jest.fn(),
+};
+
+jest.mock('next/router', () => ({
+  useRouter: () => mockedRouter,
+}));
+
+describe('MashDashboard', () => {
+  beforeEach(() => {
+    mockedRouter.query.tab = undefined;
+    mockedRouter.query.confirmation = undefined;
+  });
+
+  it('renders the dashboard default tab to be the contact tab', () => {
+    render(<MashDashboard referrals={mockedMashReferrals} />);
+
+    expect(screen.getByText(`Contact (0)`).parentElement?.className).toContain(
+      'active'
+    );
+  });
+
+  it('selects the correct tab if the url query is updated', () => {
+    mockedRouter.query.tab = 'initial-decision';
+    render(<MashDashboard referrals={mockedMashReferrals} />);
+
+    expect(
+      screen.getByText(`Initial decision (${mockedMashReferrals.length})`)
+        .parentElement?.className
+    ).toContain('active');
+  });
+
+  it('updates the selected tab when a tab is clicked on', async () => {
+    mockedRouter.query.confirmation =
+      '{"title":"test-title","test-key":"test-body"}';
+    render(<MashDashboard referrals={mockedMashReferrals} />);
+
+    expect(screen.getByText(`Contact (0)`).parentElement?.className).toContain(
+      'active'
+    );
+
+    fireEvent.click(
+      screen.getByText(`Initial decision (${mockedMashReferrals.length})`)
+    );
+
+    expect(
+      screen.getByText(`Initial decision (${mockedMashReferrals.length})`)
+        .parentElement?.className
+    ).toContain('active');
+    expect(mockedRouter.push).toHaveBeenCalledWith(
```

Comment:
Does this check that the tab switch was not done via the URL?

Reply:
Yeah, it checks that it changes through clicking a different tab rather then through the URL